pkg/apis/volumesnapshot/v1beta1/types.go

```diff
@@ -0,0 +1,330 @@
+	// Supported values are "Retain" and "Delete".
+	// "Retain" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are kept.
+	// "Delete" means that the VolumeSnapshotContent and its physical snapshot on underlying storage system are deleted.
+	// If not specified, the default value is "Retain"
+	// +kubebuilder:validation:Enum=Delete;Retain
+	// +optional
+	DeletionPolicy *DeletionPolicy `json:"deletionPolicy,omitempty" protobuf:"bytes,5,opt,name=deletionPolicy"`
+
+	// driver is the name of the CSI driver used to create the physical snapshot on
+	// the underlying storage system.
+	// This MUST be the same as the name returned by the CSI GetPluginName() call for
+	// that driver.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,1,opt,name=driver"`
+
+	// snapshotHandle is the snapshot id returned by the CSI driver in the CreateSnapshotResponse
+	// and is used as the snapshot identifier for all subsequent CSI calls.
+	// +optional
+	SnapshotHandle *string `json:"snapshotHandle,omitempty" protobuf:"bytes,2,opt,name=snapshotHandle"`
+
+	// creationTime is the timestamp when the point-in-time snapshot is taken
+	// by the underlying storage system. This timestamp is returned by the CSI
+	// driver after the snapshot is cut.
+	// The format of this field is a Unix nanoseconds time encoded as an int64.
+	// On Unix, the command `date +%s%N` returns the current time in nanoseconds
+	// since 1970-01-01 00:00:00 UTC.
+	// +optional
+	CreationTime *int64 `json:"creationTime,omitempty" protobuf:"varint,3,opt,name=creationTime"`
+
+	// restoreSize specifies the complete size of the snapshot in bytes.
+	// When restoring a volume from this snapshot, the size of the volume MUST NOT
+	// be smaller than the restoreSize if it is specified.
+	// Otherwise the restoration will fail.
+	// If this field is not set, it indicates that the size is unknown.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	RestoreSize *int64 `json:"restoreSize,omitempty" protobuf:"bytes,4,opt,name=restoreSize"`
+}
+
+// VolumeSnapshotContentStatus is the status of a VolumeSnapshotContent object
+type VolumeSnapshotContentStatus struct {
+	// readyToUse indicates if a snapshot is ready to be used as a source to restore a volume.
+	// In dynamic snapshot creation case, this field will be filled in with the value returned
+	// from CSI CreateSnapshotRequest call.
+	// For pre-existing snapshot, this field will be updated with the value returned
+	// from CSI ListSnapshotRequest if the functionality is supported by the corresponding
+	// driver. Otherwise, this field will be set to "True".
+	// Required.
+	ReadyToUse bool `json:"readyToUse" protobuf:"varint,3,opt,name=readyToUse"`
+	// error is the latest observed error during snapshot creation, if any.
+	// +optional
+	Error *VolumeSnapshotError `json:"error,omitempty" protobuf:"bytes,4,opt,name=error,casttype=VolumeSnapshotError"`
```

Comment:
4->2

Reply:
@xing-yang are u planning to address these (comments) in your next patchset ?